src/main/java/org/opensearch/ad/transport/ProfileTransportAction.java

```diff
@@ -128,6 +138,11 @@ protected ProfileNodeResponse nodeOperation(ProfileNodeRequest request) {
             // need to provide entity info for HCAD
             if (profiles.contains(DetectorProfileName.MODELS)) {
                 modelProfiles = cacheProvider.get().getAllModelProfile(detectorId);
+                modelCount = modelProfiles.size();
+                int limit = Math.min(modelToReturn, modelCount);
+                if (limit != modelCount) {
```

Comment:
Can we add a log line here if the limit is reached?

Reply:
added